misk-hibernate/src/main/kotlin/misk/hibernate/pagination/Pager.kt

```diff
@@ -0,0 +1,33 @@
+package misk.hibernate.pagination
+
+import misk.hibernate.DbEntity
+import misk.hibernate.Query
+import misk.hibernate.Session
+
+interface Pager<T> {
+  /** Returns null when there are no more pages. */
+  fun nextPage(session: Session): Page<T>?
+}
+
+/** Use a null [initialOffset] to start at the beginning. */
+fun <T : DbEntity<T>, Q : Query<T>> Q.newPager(
+  paginator: Paginator<T, Q>,
+  initialOffset: Offset? = null,
+  pageSize: Int = 50
```

Comment:
Probably the most contentious thing in the whole PR. Picking defaults here is really hard. 

This works for me!